### PR TITLE
Issue #2990: Settings form for contact module.

### PR DESCRIPTION
**Pull request by opi**
Still Work-In-Progress !

I need some help on the wording part, from a native english speaker

### CODE REVIEW

**Comment by opi** [comment]
close+reopen to re-run the tests

File: core/modules/contact/contact.admin.inc [hunk 1]
```diff
@@ -5,6 +5,23 @@
...
+ * Page callback: Contact settings.
+ *
+ * @see contact_menu()
+ */
+function contact_settings_form($form, &$form_state) {
+  $form['#config'] = 'contact.settings';
+  $form['copy'] = array(
+    '#type' => 'checkbox',
+    '#title' => t("Show the 'Send yourself a copy' checkbox"),
+    '#description' => t("Allow users to send themselves a copy on site-wide and personnal contact form"),
```
Comment (oadaeh) [inline core/modules/contact/contact.admin.inc]: The word "personal" only has one "n".
I think it is not necessary to have the "on site-wide and personnal contact form" part, as that is what the setting is for.
However, I think adding something like "of their message" would not be a bad idea.
Also, it needs a period at the end.
So, something like this: "Allow users to send themselves a copy of their message."

File: core/modules/contact/contact.module [hunk 2]
```diff
@@ -198,6 +215,18 @@ function contact_config_data() {
   return $config_data;
 }
 
+/**
+ * Show the personnal copy checkbox on forms
```
Comment (oadaeh) [inline core/modules/contact/contact.module]: The word "personal" only has one "n", and the sentence should end with a period.
I think this wording better defines the function: Determines if the 'Send yourself a copy' checkbox should be shown.

**Review by oadaeh — COMMENTED** [review]
Here is what I noticed.

**Comment by opi** [comment]
@oadaeh thanks, your concerns were addressed.

**Comment by opi** [comment]
closing to refresh the sandbox

**Comment by opi** [comment]
closing to refresh the sandbox

File: core/modules/contact/contact.admin.inc [hunk 3]
```diff
@@ -5,6 +5,55 @@
...
+  $form['fields'] = array(
+    '#type' => 'fieldset',
+    '#collapsible' => TRUE,
+    '#collapsed' => FALSE,
+    '#title' => t("Fields"),
+  );
+  $form['fields']['copy'] = array(
+    '#type' => 'checkbox',
+    '#title' => t("Show the 'Send yourself a copy' checkbox"),
+    '#description' => t("Allow users to send themselves a copy on site-wide and personnal contact form"),
```
Comment (mikemccaffrey) [inline core/modules/contact/contact.admin.inc]: Sentence should end with a period.

File: core/modules/contact/contact.module [hunk 4]
```diff
@@ -212,6 +229,18 @@ function contact_config_data() {
...
+ *
+ * @return bool
+ */
+function contact_show_personnal_copy_checkbox() {
```
Comment (quicksketch) [inline core/modules/contact/contact.module]: Same spelling issue here. `personnal` should only have a single `n`.

File: core/modules/contact/contact.admin.inc [hunk 5]
```diff
@@ -5,6 +5,55 @@
  * Admin page callbacks for the Contact module.
  */
 
+/**
+ * Page callback: Contact settings.
```
Comment (quicksketch) [inline core/modules/contact/contact.admin.inc]: As a form, this shouldn't be described as a page callback. Simply "Contact settings form." is sufficient.

File: core/modules/contact/contact.admin.inc [hunk 6]
```diff
@@ -5,6 +5,55 @@
...
+ *
+ * @see contact_menu()
```
Comment (quicksketch) [inline core/modules/contact/contact.admin.inc]: Let's include both the following:

```
@ingroup forms
@since 1.11.0
```

**Review by quicksketch — CHANGES_REQUESTED** [review]
Minor docs and naming updates. As an entirely new form and page, I think we should pend this for 1.11.0 and we need to document that in the form function docblock.

**Comment by opi** [comment]
run tests again

**Review by quicksketch — APPROVED** [review]

**Comment by quicksketch** [comment]
All tests are passing and this implementation look great! I did one more minor spelling change, otherwise everything looks great. Merged into 1.x for 1.11.0. Thanks @opi!

**Pull request merged**